Sources/Testing/Running/Runner.Context.swift

```diff
@@ -0,0 +1,204 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+extension Runner {
+  /// A type which collects the task-scoped context for a running ``Runner``
+  /// instance, the tests it runs, and other objects it interacts with.
+  ///
+  /// This type is intended for use via the task-local
+  /// ``Runner/_Context/current`` property.
+  fileprivate struct _Context: Sendable {
+    /// The runner that is running on the current task, if any.
+    var runner: Runner?
+
+    /// The test that is running on the current task, if any.
+    var test: Test?
+
+    /// The test case that is running on the current task, if any.
+    var testCase: Test.Case?
+
+    /// The context related to the runner running on the current task.
+    @TaskLocal
+    fileprivate static var current: Self = .init()
+  }
+}
+
+extension Runner {
```

Comment:
By making `Runner` the task-local value, we are hoisting much more data into task-local storage (namely the test graph.) Are we okay with that?

Reply:
I've amended my approach in the most recent commits to no longer capture the entire `Runner` in task local storage